bluesky/run_engine.py

```diff
@@ -573,18 +575,27 @@ def __call__(self, plan, subs=None, *, raise_if_interrupted=False,
               lists of callables; valid keys are {'all', 'start', 'stop',
               'event', 'descriptor'}
 
-        raise_if_interrupted : bool
-            If the RunEngine is called from inside a script or a function, it
-            can be useful to make it raise an exception to halt further
-            execution of the script after a pause or a stop. If True, these
-            interruptions (that would normally not raise any exception) will
-            raise RunEngineInterrupted. False by default.
-
         Returns
         -------
         uids : list
             list of uids (i.e. RunStart Document uids) of run(s)
         """
+        # This scheme lets us make 'plan' and 'subs' POSITIONAL ONLY, reserving
+        # all keyword arguments for user metdata.
+        sig = Signature(
```

Comment:
Can you define this outside of `__call__` so we can bind it back to call so introspection works as expected?

Reply:
How does that work? I've played around with `Signature` for a bit and I'm not sure that's possible. Got a link or a toy example?

Reply:
Signatures bind to arguments, not to functions. At least that's what `Signature.bind` does.

Reply:
https://www.python.org/dev/peps/pep-0362/#visualizing-callable-objects-signature

` wrapper.__signature__ = sig`

Reply:
Right! Done and force-pushed.